assets/gbuffer/shaders/DeferredLight.hlsl

```diff
@@ -140,91 +142,112 @@ float3 PBR(GBuffer gbuffer)
     float3 P  = gbuffer.position;
     float3 N  = gbuffer.normal;
     float3 V  = normalize(Scene.eyePosition.xyz - P);
-    float3 R  = reflect(-V, N);
-    float  ao = 0.4f;    
+    float3 R  = reflect(-V, N);    
+    float3 Lo = V;    
+    float  cosLo = saturate(dot(N, Lo));
+    //float  ao = 0.4f;    
     
 	float3 albedo   = gbuffer.albedo;    
     float roughness = gbuffer.roughness;    
     float metalness = gbuffer.metalness;
     
-    // calculate reflectance at normal incidence; if dia-electric (like plastic) use F0
-    // of 0.04 and if it's a metal, use the albedo color as F0 (metallic workflow)
-    //
-    // float3 F0 = (float3)0.04;
+    // The value for F0 is passed in from the application. If in doubt use 0.04 for 
+    // dieletric materials and scale beween 0.04 and the albedo based on the metalness
+    // value.
     //
     float3 F0 = gbuffer.F0;
     F0        = lerp(F0, albedo, metalness);
+
+
+    // Calculate direct lighting
+    float3 directLighting = (float3)0;
+    for (uint i = 0; i < Scene.lightCount; ++i) {
+        float3 Li   = normalize(Lights[i].position - P); // Incoming light direction
+        float  Lrad = Lights[i].intensity;               // Light radiance
+        float3 Lh   = normalize(Li + Lo);                // Half-vector between Li and Lo
+        float  cosLi = saturate(dot(N, Li));
+        float  cosLh = saturate(dot(N, Lh));
+
+        // Calculate Fresnel term for direct lighting 
+        float3 F = FresnelSchlick(F0, saturate(dot(Lh, Lo)));
+        // Calculate normal distribution for specular BRDF
+        float  D = DistributionGGX(cosLh, roughness);
+        // Calculate geometric attenuation for specular BRDF
+        float  G = GASmithSchlickGGX(cosLi, cosLo, roughness);
+
+        // Diffuse scattering happens due to light being refracted multiple times by a dielectric medium.
+        // Metals on the other hand either reflect or absorb energy, so diffuse contribution is always zero.
+        // To be energy conserving we must scale diffuse BRDF contribution based on Fresnel factor & metalness.
+        //
+        float3 kD = lerp(float3(1, 1, 1) - F, float3(0, 0, 0), metalness);
+
+        // clang-format off
```

Comment:
clang-format is not run on HLSL, so you can remove the directives

Reply:
Done